### PR TITLE
Fix/drawer internal state

### PR DESCRIPTION
**Pull request by ferransimon**
## MOLECULE/DRAWER
**TASK**: 

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Only
- [ ] Refactor

### Description, Motivation and Context

When pressing ESCAPE key we were not calling the onClose callback so the outer state turnes unsync with the inner state, in the current demo is not correctly appreciated because the outer state is managed as setIsOpen(!isOpen) but you have to press the button 2 times in order to be able to open the drawer again.

### CODE REVIEW

**Comment by turolopezsanabria** [comment]
great PR!
check issue #1740 please, very related to this one
thnx

File: components/molecule/drawer/src/Drawer.js [hunk 1]
```diff
@@ -20,7 +20,6 @@ const MoleculeDrawer = forwardRef(
     forwardedRef
   ) => {
     const [isOpenState, setIsOpenState] = useControlledState(isOpen) // inner state
-    const [isOpenedState, setIsOpenedState] = useState(isOpen) // transition delayed state
```
Comment (andresin87) [inline components/molecule/drawer/src/Drawer.js:23]: I think you have removed the transition dalayed behavior

Comment (ferransimon) [reply components/molecule/drawer/src/Drawer.js:23]: How can I reproduce that? I do not see any strange behavior in the demo

**Review by turolopezsanabria — COMMENTED** [review]
tested in safari and chrome and works perfect
Did you have the chance to address @andresin87 comments?

**Review by andresin87 — APPROVED** [review]

**Comment by andresin87** [comment]
this issue will not solve #1740

**Comment by turolopezsanabria** [comment]
> this issue will not solve #1740

ok, fine for me

**Pull request merged**